src/compiler/checker.ts

```diff
@@ -4196,9 +4317,10 @@ namespace ts {
         }
 
         // Get type from reference to class or interface
-        function getTypeFromClassOrInterfaceReference(node: TypeReferenceNode | ExpressionWithTypeArguments, symbol: Symbol): Type {
+        function getTypeFromClassOrInterfaceReference(node: TypeReferenceNode | ExpressionWithTypeArguments | JSDocTypeReference, symbol: Symbol): Type {

```

Comment:
It might be more readable just to create a type alias for "TypeReferenceNode | ExpressionWithTypeArguments | JSDocTypeReference". It's used repeatedly (7 times) in the following code.